Consul/Client_GetRequests.cs

```diff
@@ -79,8 +79,13 @@ public async Task<QueryResult<TOut>> Execute(CancellationToken ct)
             result.StatusCode = response.StatusCode;
             ResponseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
 
-            if (response.StatusCode != HttpStatusCode.NotFound && !response.IsSuccessStatusCode)
+            var isSpecialStatusCode = response.StatusCode == HttpStatusCode.ServiceUnavailable || (int)response.StatusCode == 429;
+
+            if (!response.IsSuccessStatusCode && !isSpecialStatusCode && (
+                (response.StatusCode != HttpStatusCode.NotFound && typeof(TOut) != typeof(TxnResponse)) ||
```

Comment:
Comparing the type to `TxnResponse` is not correct, the actual type we care about here is `LocalServiceHealth[]` and `LocalServiceHealth`.
On the other hand, instead of checking the type we can check the actual endpoint URL, to write a special case for it e.g.:
```
            var isSpecialCase =
                ((int)response.StatusCode == 503 && Endpoint.StartsWith("v1/agent/health/service/name/", StringComparison.OrdinalIgnoreCase)) ||
                ((int)response.StatusCode == 429 && Endpoint.StartsWith("v1/agent/health/service/name/", StringComparison.OrdinalIgnoreCase)) ||
                ((int)response.StatusCode == 503 && Endpoint.StartsWith("v1/agent/health/service/id/", StringComparison.OrdinalIgnoreCase)) ||
                ((int)response.StatusCode == 429 && Endpoint.StartsWith("v1/agent/health/service/id/", StringComparison.OrdinalIgnoreCase))
```

I think that way is more explicit and easier to understand;

Reply:
I agree 😄